### ISSUE TITLE
Edlib admin: Make login happen before page load

### ISSUE THREAD

**Issue by Kamelon7** [cerpus/Edlib#1507]
At present, you can access the admin page as long as you have the URL. You can then log in afterwards. Login prompt should happen before the admin page loads to strengthen security.

**Comment by emmachughes** [comment]
I don't think there's anything to be done here. The endpoints for the admin page require signed tokens with a particular payload to access, which can only be obtained by logging in with admin credentials. The admin area is a single-page application (SPA), so the HTML received by the client is always free of any sensitive data, only populated on the browser-side with data from the restricted endpoints while logged in. Also due to the admin area being an SPA, the endpoints could not be protected by stateful security mechanisms such as [basic access authentication](https://en.wikipedia.org/wiki/Basic_access_authentication) or login sessions like a traditional, server-side rendered page. While there might be an argument for security by obscurity by not making the admin area's frontend code available to the public, there is nothing to hide due to Edlib's open source nature.

In short: while it seems counter-intuitive to security to leave the admin area available prior to logging in, nothing would actually change if we attempted to hide it away.

For these reasons, I am closing the issue.
